src/metabase/formatter.clj

```diff
@@ -122,12 +122,15 @@
                                (streaming.common/global-type-settings col viz-settings)
                                (catch Exception _e
                                  (streaming.common/global-type-settings (dissoc col :base_type :effective_type) viz-settings)))
-        col-id               (or col-id (second field-ref))
+        ref-type             (first field-ref)
+        col-id-or-name       (or col-id (second field-ref))
```

Comment:
maybe @bshepherdson can confirm but there might be a metabase-lib way to extract these values from the field ref?

Reply:
The query and field ref here are legacy MBQL, not the lib's PMBQL. I agree it's a code smell to be reaching into these structures in many hundreds of places in the codebase. But that ship has not just sailed, it's circumnavigated the globe, discovered a new continent, and wrecked on its shores.

So I think this is fine. All legacy refs have the form `[ref-type identifying-thing & maybe-more-stuff]`.